openedx/core/djangoapps/content/search/documents.py

```diff
@@ -275,3 +277,36 @@ def searchable_doc_for_course_block(block) -> dict:
     doc.update(_fields_from_block(block))
 
     return doc
+
+
+def searchable_doc_for_collection(collection) -> dict:
+    """
+    Generate a dictionary document suitable for ingestion into a search engine
+    like Meilisearch or Elasticsearch, so that the given collection can be
+    found using faceted search.
+    """
+    doc = {
+        Fields.id: collection.id,
+        Fields.type: DocType.collection,
+        Fields.display_name: collection.title,
+        Fields.description: collection.description,
+        Fields.created: collection.created.timestamp(),
+        Fields.modified: collection.modified.timestamp(),
+        # Add related learning_package.key as context_key by default.
+        # If related contentlibrary is found, it will override this value below.
+        # Mostly contentlibrary.library_key == learning_package.key
+        Fields.context_key: collection.learning_package.key,
+    }
+    # Just in case learning_package is not related to a library
+    if hasattr(collection.learning_package, 'contentlibrary'):
```

Comment:
Is this actually the way to check if a learning package has an associated content library? It doesn't seem right..
Could you add a test for the case where a collection is associated with a learning package that is _not_ associated with a content library, so we can be sure this logic is correct?

Reply:
@pomegranited That is a good point. I am not really sure if we have a scenario where a learning_package is not related to a library, maybe a course?

Reply:
Ya, within the context of openedx, a collection _should_ always be related to a learning_package with a content library (courses don't have collections). I just think the `hasattr` mechanism is weird, and may always be true, even if `collection.learning_package.contentlibrary == None`.

Reply:
@pomegranited Ohh, I added a test with a learning package that is not associated with a library. The `hasattr` mechanism worked but like you said it is weird and may not work in case the attr is none, so updated to make use of `try..catch` DoesNotExist exception.